components/properties/src/maps.rs

```diff
@@ -14,55 +14,25 @@
 
 use crate::error::PropertiesError;
 use crate::provider::*;
+#[cfg(doc)]
 use crate::*;
-use icu_codepointtrie::TrieValue;
 use icu_provider::prelude::*;
-use BidiClass;
-
-/// TODO(#1239): Finalize this API.
-pub type CodePointMapResult<T> =
-    Result<DataPayload<UnicodePropertyMapV1Marker<T>>, PropertiesError>;
-
-fn get_cp_map<D, T>(provider: &D, key: ResourceKey) -> CodePointMapResult<T>
-where
-    D: DynProvider<UnicodePropertyMapV1Marker<T>> + ?Sized,
-    T: TrieValue,
-{
-    let resp: DataResponse<UnicodePropertyMapV1Marker<T>> =
-        provider.load_payload(key, &Default::default())?;
-
-    let property_payload: DataPayload<UnicodePropertyMapV1Marker<T>> = resp.take_payload()?;
-    Ok(property_payload)
-}
 
 macro_rules! make_map_property {
     (
-        // currently unused
-        property: $prop_name:expr;
-        // currently unused
-        marker: $marker_name:ident;
-        value: $value_ty:path;
-        key: $key_name:expr;
-        func:
+        $marker:ident,
```

Comment:
issue: please don't remove or change the macro inputs (feel free to add new ones if you need them). this was done deliberately to make room for the properties changes, so that the refactors can be done  efficiently, If things are removed I may have to go through and re-add things later.

The GeneralCategoryProperty stuff, for example, is important and will be used in the future, it's just unused right now. The "marker" argument to this macro is _not_ a resource marker, it's a property marker, a concept which has not yet been introduced.

Ideally it would be nice if the changes here could be split out into a separate PR, too, it's kinda hard to review.

Reply:
Made #1862 and rebased on that

Reply:
thanks!